pages/api/[entity_type_slug]/[slug].js

```diff
@@ -0,0 +1,51 @@
+import { withSession } from "@klaudsol/commons/lib/Session";
+import RecordNotFound from "@klaudsol/commons/errors/RecordNotFound";
+import { OK, NOT_FOUND } from "@klaudsol/commons/lib/HttpStatuses";
+import { setCORSHeaders, handleRequests } from "@klaudsol/commons/lib/API";
+import { assertUserCan } from '@klaudsol/commons/lib/Permissions';
+import { createHash } from "@/lib/Hash";
+import { readContents } from '@/lib/Constants';
+import Entity from "@/backend/models/dynamodb/Entity";
+import { 
+    formatAttributesData, 
+    formatContentData, 
+    getAttributeMap, 
+    getEntityVariant 
+} from "@/utils/dynamodb/formatResponse";
+import { DYNAMO_DB_ID_SEPARATOR } from "@/constants";
+
+export default withSession(handleRequests({ get }));
+
+async function get(req, res) {
+    await assertUserCan(readContents, req);
+
+    const { entity_type_slug, slug } = req.query;
+
+    const rawData = await Entity.find({ entity_type_slug, slug });
+    const rawMetadata = await Entity.whereMetadata({ entity_type_slug });
+    if (rawMetadata.Items.length === 0) throw new RecordNotFound();
+    if (rawData.Items.length === 0) throw new RecordNotFound();
+
+    const entityIdRaw = rawData.Items[0].PK.S;
+    const entity_type_id = entityIdRaw.split(DYNAMO_DB_ID_SEPARATOR)[1];
```

Comment:
According to the Marcia Villalba and Alex De Brie, it's better to maintain a field copy that contains the value (e.g.  `rawData.Items[0].type.S`), rather than parse the primary keys.